src/lib.rs

```diff
@@ -16,3 +16,31 @@ pub mod databases;
 pub mod jobs;
 pub mod api;
 pub mod protocol;
+
+
+#[macro_use]
+extern crate lazy_static;
+
+pub mod keys {
+    lazy_static! {
+        pub static ref BLOWFISH_KEY: blowfish::Blowfish<byteorder::LE> =
+            <blowfish::BlowfishLE as crypto_common::KeyInit>::new(
+                &<blowfish::BlowfishLE as crypto_common::KeyInit>::generate_key(
+                    rand::rngs::ThreadRng::default()
+                )
+            );
+        pub static ref TEST_BLOWFISH_KEY: blowfish::Blowfish<byteorder::LE> =
+            <blowfish::BlowfishLE as crypto_common::KeyInit>::new(
+                &<blowfish::BlowfishLE as crypto_common::KeyInit>::generate_key(
+                    <rand::rngs::StdRng as rand::SeedableRng>::seed_from_u64(0)
+                )
+            );
+        pub static ref SECRET_KEY: [u8; 32] = rand::Rng::gen(&mut rand::rngs::ThreadRng::default());
+    }
+}
+
+pub fn initialize_keys() {
+    // Initialize Torrust Blowfish Crypto Key
+    lazy_static::initialize(&keys::BLOWFISH_KEY);
+    lazy_static::initialize(&keys::TEST_BLOWFISH_KEY);
+}
```

Comment:
> 1. I think I would delay this initialization until needed. As I told @WarmBeer I tend to prefer keeping things as local as possible. These keys are only being used right now by the UDP tracker, and you can even not run it when you start the main app. We could move up as long as other parts of the application start using them. In fact, we could even consider using different keys.

Generally, we want these keys to be initialized as early as possible. Global static values can be accessed by anyone at anytime, so we want to make sure they are ready before we do anything else.

For this project the overheard is so low that I don't see the problem about just initializing and forgetting.
- The problem I see is that this memory might be maliciously accessed (for example swapped to disk and accessed elsewhere), however this is less of a problem, as no persistent credentials are stored here; when this application is restart new keys are generated from random.

I think that we put in a todo to replace it with a double [arc_swap](https://docs.rs/arc-swap/latest/arc_swap/), that rotates these these keys out every hour or so.  However that is more complex than this simple solution. 

> 2. I would keep this mod inside the udp::connection.

This is unfortunately not possible. Lazy Static demands to be in the root of the crate.

> 3. I would prefer to pass the value down instead of using a global variable.

If this were some very large project with lots of crypto systems who's contexts should programmatically be kept separate from each other, this project doesn't have those design requirements.

> 4. I would prefer to keep the test code separated from the production code. I suppose that forces you to inject the value instead of using a global one.

I share the same preference; however this solution is most simple and works.

Reply:
I much prefer to initialize the secret on UdpServer::start() as it isn’t needed anywhere else. The blowfish key can then be a field of PacketHandler.

In the future, we can use https://doc.rust-lang.org/cargo/reference/features.html to include/exclude UDP and HTTP code. It would be nice if all relevant code of these features is only part of the crate it is destined for.

This way, if someone wants to use Torrust without the UDP feature enabled. They can automatically exclude all crates/dependencies that are only needed for UDP.